examples/c-guest/host/src/main.rs

```diff
@@ -0,0 +1,26 @@
+use risc0_zkvm::{compute_image_id, default_prover, ExecutorEnv};
+use std::fs;
+
+fn main() -> anyhow::Result<()> {
+    // Load built gcc program and compute it's image ID.
+    let consensus_elf = fs::read("./guest/out/main")?;
+    let consensus_id = compute_image_id(&consensus_elf)?;
```

Comment:
This seems like a dangerous example to set. Ideally the image_id is computed at build time.

Reply:
agree, but very bloated to do at compile time. Would a lazy static be sufficient here? Or should I resolve it through a build script/proc macro?

Reply:
But I guess practically for this example, should this happen in a build script that builds some minimal crate with those types or in a proc macro?

lazy_static now that I think about it wouldn't alleviate your concern, as it's still being retrieved at runtime (once).

Reply:
We can iterate on this, it doesn't need to block landing it. But I do think we need to work on how do we generate the image_id at build time generally. I think running it from the Makefile would make sense to me.

Reply:
Added it as a TODO here, to at least mark it if anyone building off it before we update 👍